benchmark/stats/stats.go

```diff
@@ -30,28 +30,130 @@ import (
+			s += "Network: Longhaul \n"
+			isNetwork = true
+		}
+	}
+	if !isNetwork {
+		if noneEmptyPos[1] {
+			s += fmt.Sprintf("%slatency%s%s%s", prefix, linker, f.Latency.String(), suffix)
+		}
+		if noneEmptyPos[2] {
+			s += fmt.Sprintf("%skbps%s%#v%s", prefix, linker, f.Kbps, suffix)
+		}
+		if noneEmptyPos[3] {
+			s += fmt.Sprintf("%sMTU%s%#v%s", prefix, linker, f.Mtu, suffix)
+		}
+	}
+	if noneEmptyPos[4] {
+		s += fmt.Sprintf("%sCallers%s%#v%s", prefix, linker, f.MaxConcurrentCalls, suffix)
+	}
+	if noneEmptyPos[5] {
+		s += fmt.Sprintf("%sreqSize%s%#vB%s", prefix, linker, f.ReqSizeBytes, suffix)
+	}
+	if noneEmptyPos[6] {
+		s += fmt.Sprintf("%srespSize%s%#vB%s", prefix, linker, f.RespSizeBytes, suffix)
+	}
+	if noneEmptyPos[7] {
+		s += fmt.Sprintf("%sCompressor%s%t%s", prefix, linker, f.EnableCompressor, suffix)
+	}
+	return s
+}
+
 type percentLatency struct {
-	percent string
-	value   time.Duration
+	Percent string
+	Value   time.Duration
 }
 
-type latencyResults struct {
-	latency []percentLatency
+// BenchResults records features and result of a benchmark.
+type BenchResults struct {
+	RunMode           string
+	Features          Features
+	Latency           []percentLatency
+	Operations        int
+	NsPerOp           int64
+	AllocedBytesPerOp int64
+	AllocsPerOp       int64
+	SharedPosion      []bool
 }
 
-// LatencyString output latency stats as the format as time + unit.
-func (stats *Stats) LatencyString() string {
+// SetBenchmarkResult sets features of benchmark and basic results.
+func (stats *Stats) SetBenchmarkResult(mode string, features Features, o int, ns, allocdBytes, allocs int64, sharedPos []bool) {
+	stats.result.RunMode = mode
+	stats.result.Features = features
+	stats.result.Operations = o
+	stats.result.NsPerOp = ns
+	stats.result.AllocedBytesPerOp = allocdBytes
+	stats.result.AllocsPerOp = allocs
+	stats.result.SharedPosion = sharedPos
+}
+
+// GetBenchmarkResults returns the result of the benchmark including features and result.
+func (stats *Stats) GetBenchmarkResults() BenchResults {
+	return stats.result
+}
+
+// BenchString output latency stats as the format as time + unit.
+func (stats *Stats) BenchString() string {
 	stats.maybeUpdate()
 	s := stats.result
-	var res string
-	if len(s.latency) != 0 {
-		var statsUnit = s.latency[0].value
+	res := s.RunMode + "-" + s.Features.String() + ": \n"
+	if len(s.Latency) != 0 {
+		var statsUnit = s.Latency[0].Value
 		var timeUnit = fmt.Sprintf("%v", statsUnit)[1:]
-		for i := 1; i < len(s.latency); i++ {
-			res += fmt.Sprintf("  %*s %*s  ", 12,
-				strconv.FormatFloat(float64(s.latency[i].value)/float64(statsUnit), 'f', 4, 64), 2, timeUnit)
+		for i := 1; i < len(s.Latency); i++ {
+			res += fmt.Sprintf("%s_Latency: %s %s \t", s.Latency[i].Percent,
+				strconv.FormatFloat(float64(s.Latency[i].Value)/float64(statsUnit), 'f', 4, 64), timeUnit)
 		}
 	}
+	res += fmt.Sprintf("Count: %s \t", strconv.Itoa(s.Operations))
```

Comment:
Just print the value directly:

res += fmt.Sprintf("Count: %v \t", s.Operations)  // or %d works too

Reply:
Done.